### ISSUE TITLE
Presentation mode in a dual display setting

### ISSUE THREAD

**Issue by parrenin** [ONLYOFFICE/DesktopEditors#285]
**Do you want to request a *feature* or report a *bug*?**
feature
**What is the current behavior?**
In presentation mode with a dual display, the presentation view goes full screen but only on the display where OnlyOffice is located.
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
When in presentation mode with a dual display, the second display should show the presentation in full screen, and the primary display should show useful information like a timer, a preview of the next slide, the notes, etc. 
**Which versions of DesktopEditors, and which OS are affected by this issue? Did this work in previous versions of DesktopEditors?**
5.4.1.33

**Comment by Rita-Bubnova** [comment]
Hello, @parrenin.
Thank you for your enhancement.
I can confirm - this is known issue 39337 in our private tracker.

**Comment by parrenin** [comment]
This request is still valid in 7.2.0.
I hope this can be worked on soon.
This is for me a blocking issue to use OO for presentations.

**Comment by parrenin** [comment]
I checked today with 7.3.0 and I did not find any presentation mode with a dual-display setting, on neither Windows, nor Linux (using GNOME/X11, GNOME/Wayland, KDE/X11 or KDE/Wayland).
The presentation just shows in full-screen on the primary display, no presenter view.
@shu0406: which OS do you use? How did you trigger the presenter view?

**Comment by shu0406** [comment]
@parrenin : I use Zorin 16.2. with OO 7.2.1.. 
There is a presenter view, but it opens in the backround of the primary display, so maybe you did not see it. 

Imo your are right, the default mode should be:
**start presentation always in presenter view AND with presenter view on primary display AND the presentation on extended display.**

That is the most common use case since I (and everybody I know) just uses PowerPoint/Presentations if there is a audience that looks at a big extended screen which I have in my back while watching notes on the small primary screen in front of me.

**Comment by parrenin** [comment]
@shu0406: OK, I now see the presenter mode on Linux/Gnome when putting the OO window on the secondary screen and starting the presenter mode from there. It works both on Gnome/Wayland and Gnome/X11.
But it does not work on KDE-plasma, neither with X11 nor with Wayland.
The presentation always shows up on the primary screen, even though the OO Window is on the external screen when you start the presenter mode. I am using KDE-plasma 5.26.90 (5.27beta) from Debian Bookworm.

**Comment by parrenin** [comment]
OK, now I see the presenter window on KDE-Plasma, by switching windows with CTRL-TAB.
Indeed, it is always hidden behind the presentation.
What is strange on Plasma is that the slide always starts on my laptop screen, even if it is not configured as the primary screen.
So not very useful, quite the contrary of what you would expect.

**Comment by theowoo** [comment]
@parrenin Have you tried switching the relative positions of the two screens in  display settings in KDE? Does it affect where the presentation is started?

**Comment by bdoublet91** [comment]
Hi,
Any update on this ?
Little complicated for presentation.
However thanks for your job, the application is awesome :)

**Comment by parrenin** [comment]
I think this issue is fixed and can be closed now.

**Comment by maruacat** [comment]
Hello @parrenin 
I agree, the feature was successfully released so we can close this issue.

**Comment by Protarios** [comment]
I have a three monitor setup in Linux (KDE Plasma Wayland) and it does not work as intended.
Presentation does not start on the second screen, but always on the internal laptop display that has lowest priority in screen settings.
This makes the presenter unusable as the laptop screen is the screen you definitely do not want the presentation to be shown on.

**Comment by maruacat** [comment]
> I have a three monitor setup in Linux (KDE Plasma Wayland) and it does not work as intended. Presentation does not start on the second screen, but always on the internal laptop display that has lowest priority in screen settings. This makes the presenter unusable as the laptop screen is the screen you definitely do not want the presentation to be shown on.

Hello @Protarios 
We tried to reproduce your problem, but unfortunately we were unable to. Therefore, could you provide us with additional information:
- What version of the editor are you using?
- What OS do you have?
- Is your problem reproducible without Wayland?
- Does the problem occur with 2 monitors, i.e. one speaker monitor and one listener monitor?
- Are there any additional settings in your OS for playing data on additional monitors?
Thank you for your message.

**Comment by b2cc** [comment]
Hi @maruacat ,

I can confirm that, at lest under KDE/Wayland, this doesn't work as intended. I have no Xorg setup anymore to test if this is Wayland-only though.

E.g. I have a 3-monitor setup, with the left monitor being turned 90° (see below). The center monitor has the highest priority, followed by the right and left monitors. However, the presentation always starts on 90°-turned left monitor. I haven't found any way to force the presentation window to another screen. Even If I make the application start on the middle or right screen, the presentation always goes to the left screen.

Thinking of how many possible configurations exist out there I believe it would be best if we had the possibility to chose the screens for the presentation and the notes before starting the presentation (e.g. displaying a small window with the detected screen layout and being able to select the screens by ourselves).

Screen layout and OO version:
![onlyoffice_presentation](https://github.com/ONLYOFFICE/DesktopEditors/assets/24914972/8cd546bf-c3c7-40d4-a233-e15271c70c5f)
![onlyoffice_presentation2](https://github.com/ONLYOFFICE/DesktopEditors/assets/24914972/3a7b04a7-af55-4d5c-8e3d-a012c94df7b6)

**Comment by maruacat** [comment]
> Hi @maruacat ,
> 
> I can confirm that, at least under KDE/Wayland, this doesn't work as intended. I have no Xorg setup anymore to test if this is Wayland-only though.
> 
> E.g. I have a 3-monitor setup, with the left monitor being turned 90° (see below). The center monitor has the highest priority, followed by the right and left monitors. However, the presentation always starts on 90°-turned left monitor. I haven't found any way to force the presentation window to another screen. Even If I make the application start on the middle or right screen, the presentation always goes to the left screen.
> 
> Thinking of how many possible configurations exist out there I believe it would be best if we had the possibility to chose the screens for the presentation and the notes before starting the presentation (e.g. displaying a small window with the detected screen layout and being able to select the screens by ourselves).
> 
> Screen layout and OO version: ![onlyoffice_presentation](https://private-user-images.githubusercontent.com/24914972/319485169-8cd546bf-c3c7-40d4-a233-e15271c70c5f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyMzc3MTEsIm5iZiI6MTcxMjIzNzQxMSwicGF0aCI6Ii8yNDkxNDk3Mi8zMTk0ODUxNjktOGNkNTQ2YmYtYzNjNy00MGQ0LWEyMzMtZTE1MjcxYzcwYzVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA0VDEzMzAxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4MzUxY2I4MTVlMTUzMzNmMzViMjgwNDBhM2MwMTBlN2IzM2M3MDY3YjA1OGIwMGU0MzZlYWRjZGMwMGIxNzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.nfP49ifwjjUjjalSiTuJSOUmLHiJcc9tHVyIatDntIM) ![onlyoffice_presentation2](https://private-user-images.githubusercontent.com/24914972/319485184-3a7b04a7-af55-4d5c-8e3d-a012c94df7b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIyMzc3MTEsIm5iZiI6MTcxMjIzNzQxMSwicGF0aCI6Ii8yNDkxNDk3Mi8zMTk0ODUxODQtM2E3YjA0YTctYWY1NS00ZDVjLThlM2QtYTAxMmM5NGRmN2I2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA0VDEzMzAxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjNzE1NjY3ZGZkOWMwYWJlZTRjNGEyN2QxMTA1NzM0YjAxMjY4M2FkMTRiOTRhOTczMDY0MGRhYTAyOGJlMTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2dSm1XwS-8u0IUuNMN1cqfH0A3rLEmH-j8nmVWGzqYc)

Hello @b2cc 
Thank you very much for your answer. I think that we need a separate issue for your problem in order to understand your problem in more detail, because in the original issue the problem was with two monitors. So could you please create a separate issue. And please write the name of your OS when you will create a new issue, so that we can try to reproduce the problem with the conditions that you have.

**Comment by pbrada** [comment]
Hello, I want to confirm that the behaviour is not as expected (and reported above as fixed): the presenter view works only on secondary display, and the fullscreen presentation is shown on main display.  This is on Desktop: Xfce v: 4.18.1 Distro: Linux Mint 22 Wilma, Onlyoffice Desktop version 8.1.0.169 (deb).

**Comment by cyberwolfie1** [comment]
I also have this issue on KDE Plasma under Wayland. Version: 8.2.0.143

**Comment by davidgood** [comment]
Problem exists in OO 8.2.1.31 flatpak, Fedora 41 with KDE Plasma Wayland.

No option for second monitor, which in this case is a projector connected via HDMI, as presentation display.  I start a presentation and only seems to recognize the laptop display.

**Comment by ElenaMaaya** [comment]
> This is still an issue in the lates version of OO

Hello, @dajomu1 !
Thank you for choosing our product! We appreciate your support and are here to help you.
To assist you more effectively, we kindly ask you to provide the following information:

1. A detailed description of the issue you are experiencing.
2. The operating system you are using.
3. The version of the operating system.

Thank you for your cooperation!

**Comment by dajomu1** [comment]
> > This is still an issue in the lates version of OO
> 
> Hello, [@dajomu1](https://github.com/dajomu1) ! Thank you for choosing our product! We appreciate your support and are here to help you. To assist you more effectively, we kindly ask you to provide the following information:
> 
> 1. A detailed description of the issue you are experiencing.
> 2. The operating system you are using.
> 3. The version of the operating system.
> 
> Thank you for your cooperation!

Hi,

1. There is no way to choose which monitor to display the presentation. 
2. I am using Kubuntu 24.10
3. latest version of OO. 

I have read that one can move OO to second monitor, start the presentation and drag OO back to primary monitor. This does not work and this is a lazy ass solution for the developers. Why can pther open source program like libre office manage this? It is open source so check the code for the solution. this is basically the only issue i have so far with OO and I really hope this will be fixed.

**Comment by ElenaMaaya** [comment]
> 
> Hi,
> 
> 1. There is no way to choose which monitor to display the presentation.
> 2. I am using Kubuntu 24.10
> 3. latest version of OO.
> 
> I have read that one can move OO to second monitor, start the presentation and drag OO back to primary monitor. This does not work and this is a lazy ass solution for the developers. 

@dajomu1 , thank you for your response. We will investigate the issue you reported and get back to you as soon as possible.
